src/frontend/src/lib/components/docs/DocForm.svelte

```diff
@@ -0,0 +1,272 @@
+
+<script lang="ts">
+  import IconNew from '$lib/components/icons/IconNew.svelte';
+	import { DocFieldTypeEnum, type DocField } from '$lib/types/doc-form';
+	import IconDelete from '../icons/IconDelete.svelte';
+  import { busy } from '$lib/stores/busy.store';
+	import { setDoc } from '$lib/api/satellites.api';
+  import { toasts } from '$lib/stores/toasts.store';
+	import { RULES_CONTEXT_KEY, type RulesContext } from '$lib/types/rules.context';
+	import type { Principal } from '@dfinity/principal';
+	import { createEventDispatcher, getContext } from 'svelte';
+	import { generateUniqueDocID } from '$lib/utils/doc.utils';
+	import { isNullish } from '$lib/utils/utils';
+  import { i18n } from '$lib/stores/i18n.store';
+	import { toArray, toNullable } from '$lib/utils/did.utils';
+	import { DATA_CONTEXT_KEY, DataStoreStateEnum, type DataContext } from '$lib/types/data.context';
+	import type Doc from './Doc.svelte';
+
+  const { store, reload }: RulesContext = getContext<RulesContext>(RULES_CONTEXT_KEY);
+  const { store: docsStore }: DataContext<Doc> = getContext<DataContext<Doc>>(DATA_CONTEXT_KEY);
+
+
+  let state: DataStoreStateEnum | undefined
+  $: state = $docsStore.state
+  let collection: string | undefined;
+	$: collection = $store.rule?.[0];
+
+	let satelliteId: Principal;
+	$: satelliteId = $store.satelliteId;
+
+  let docId: string = '';
+  let fields: DocField[] = [
+    {
+      name: '',
+      fieldType: DocFieldTypeEnum.boolean,
+      value: false,
+    }
+  ]
+
+  const onAddFieldButtonPressed = () => {
+    fields = [
+      ...fields,
+      {
+        name: '',
+        fieldType: DocFieldTypeEnum.boolean,
+        value: false,
+      }
+    ]
+  }
+
+  const onDeleteFieldPressed = (index: number) => {
+    if (fields.length > 1) {
+      fields = fields.filter((_, i) => i !== index)
+    }
+  }
+
+  let isFormValid = false
+  $: isFormValid = !!docId && !!fields.find((field) => !!field.name && !isNullish(field.value) && field.value !== '')
```

Comment:
can you use `isNullish` and `nonNullish` instead of `!!`?

for sure it works, I just use those helper so for consistency would be nice